src/main/java/com/gmail/goosius/siegewar/command/SiegeWarAdminCommand.java

```diff
@@ -226,30 +226,48 @@ private void parseSiegeWarImmunityCommand(CommandSender sender, String[] args) {
 				Messaging.sendErrorMsg(sender, Translation.of("msg_err_not_registered_1", args[1]));
 				return;
 			}
-			long durationMillis = (long)(Long.parseLong(args[2]) * TimeMgmt.ONE_HOUR_IN_MILLIS);
-			TownMetaDataController.setSiegeImmunityEndTime(town, System.currentTimeMillis() + durationMillis);
-			TownyMessaging.sendPrefixedTownMessage(town, Translation.of("msg_set_siege_immunities_town", args[1], args[2]));
-			Messaging.sendMsg(sender, Translation.of("msg_set_siege_immunities_town", args[1], args[2]));
-
-		} else if (args.length >= 3 && args[0].equalsIgnoreCase("nation")) {
+			if (args[2].equalsIgnoreCase("permanent")) {
+				TownMetaDataController.setSiegeImmunityEndTime(town, -1l);
+				TownyMessaging.sendPrefixedTownMessage(town, Translation.of("msg_swa_town_unsiegeable_set", args[1]));
+				Messaging.sendMsg(sender, Translation.of("msg_swa_town_unsiegeable_set", args[1])); //this one

```

Comment:
- The permanent option is just another option on the menu, so it would be best if the confirmation message had the same format for all option
- For example, if an admin sets it by hour, they see `"Siege immunity for town x set to 6 hours."`
- So if an admin sets it to permanent, they should see the same format:  `Siege immunity for town x set to Permanent'`
- To do this, I suggest setting up your lang strings like this:
  - msg_hours: 'hours'  ........ this string is already in the Towny  english.yml 
  - msg_permanent: 'permanent' .............add this string to sw english.yml
  - msg_set_siege_immunities_town:  "Siege immunity for town %s set to %s." ..... update existing string
  - msg_set_siege_immunities_nation:  "etc..."  ..... update existing string
  - msg_set_siege_immunities_all:  "etc.."   ..... update existing string